### PR TITLE
User defined model

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

**Comment by raphaelsaavedra** [comment]
Looks fine, but this test needs to be removed:

`@test_throws ErrorException StateSpaceModel(y, Z, T, R, dim, "asdf")`

File: src/structures.jl [hunk 1]
```diff
@@ -41,19 +41,35 @@ struct StateSpaceModel
...
-        if mode != "time-variant" && mode != "time-invariant"
-            error("mode should be either 'time-variant' or 'time-invariant'.")
+    function StateSpaceModel(y::Matrix{Float64}, Z::Vector{Matrix{Float64}}, T::Matrix{Float64}, R::Matrix{Float64})
+        
+        # Validate StateSpaceDimensions
+        ny, py = size(y)
+        pz, mz = size(Z[1])
+        mt, mt = size(T)
+        mr, rr = size(R)
+        if !((mz == mt == mr) && (pz == py))
```
Comment (raphaelsaavedra) [inline src/structures.jl]: An additional check such as `all([size(Z[i]) == size(Z[1]) for i = 2:length(Z)])` would be interesting to cover the case where Z has non-constant dimensions

Comment (guilhermebodin) [reply src/structures.jl]: Ok

File: src/structures.jl [hunk 2]
```diff
@@ -43,6 +43,10 @@ struct StateSpaceModel
 
     function StateSpaceModel(y::Matrix{Float64}, Z::Vector{Matrix{Float64}}, T::Matrix{Float64}, R::Matrix{Float64})
         
+        # Check if Z has the same dimensions for all instants
+        if !all([size(Z[i]) == size(Z[1]) for i = 2:length(Z)])
```
[the remark this answers is not in the document]
Comment (guilhermebodin) [reply src/structures.jl]: merging

**Pull request merged**